service/amount_out.go

```diff
@@ -57,6 +57,10 @@ func getOutAmountInAToken(ctx context.Context, it models.OrderIter, inAmountB de
 	var order *models.Order
 	for it.HasNext() && inAmountB.IsPositive() {
 		order = it.Next(ctx)
+		if order == nil {
+			logctx.Error(ctx, "order::it.Next() returned nil")
```

Comment:
Can you update the Error log to Warn then as this is not an unexpected error, and change the error? You could do models.ErrInsufficientLiquidity for example?

Reply:
i handled it in the calling func to return insufficient liqidity
`it = s.orderBookStore.GetMaxBid(ctx, symbol)
		if it == nil {
			logctx.Error(ctx, "GetMaxBid failed no orders in iterator")
			return models.QuoteRes{}, models.ErrIterFail
		}
		if !it.HasNext() {
			logctx.Warn(ctx, "GetMaxBid failed no orders in iterator")
			return models.QuoteRes{}, models.ErrInsufficientLiquity
		}`

Reply:
so i leave this code as it was, since it is indeed not expected